src/Analyzers/CSharp/CodeFixes/SimplifyLinqExpression/CSharpSimplifyLinqExpressionCodeFixProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.SimplifyLinqExpression;
+
+namespace Microsoft.CodeAnalysis.CSharp.SimplifyLinqExpression
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal sealed class CSharpSimplifyLinqExpressionCodeFixProvider : AbstractSimplifyLinqExpressionCodeFixProvider
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpSimplifyLinqExpressionCodeFixProvider()
+        {
+        }
+
+        protected override SyntaxNode? GetIdentifierName(SyntaxNode node)
+            => (node.DescendantNodesAndSelf().OfType<InvocationExpressionSyntax>().First().Expression as MemberAccessExpressionSyntax)?.Name;
```

Comment:
something isn't ok here.  `node.DescendantNodesAndSelf()` is too broad.  We should have strongly matched a specific pattern and should have the explicit nodes we're trying to grab data from.

Reply:
(issue arises for the reset of hte methods).

Reply:
I can use the additional locations to pull this from instead.